src/acp/acp.ts

```diff
@@ -333,11 +333,17 @@ async function fetchAcr(
     // an ACL is advertised, we can still fetch its metadata — if that indicates
     // that it's actually an ACP Access Control Resource, then we can fetch that
     // instead.
-    const aclResourceInfo = await getResourceInfo(
-      resource.internal_resourceInfo.aclUrl,
-      options
-    );
-    if (isAcr(aclResourceInfo)) {
+    let aclResourceInfo;
+    try {
+      aclResourceInfo = await getResourceInfo(
+        resource.internal_resourceInfo.aclUrl,
+        options
+      );
+    } catch (e) {
+      // if the above fails, an ACL cannot be found and we proceed.
```

Comment:
I think it would be helpful to expand this comment a bit, to recontextualize why that's a correct behavior, rather than describing what the code does. For instance, remind that both ACL and ACR will be discovered through the same header, and that an additional lookup is required for checking the type of the resource, and that the absence of the resource is only possible in the case of an ACL, because in the ACR system the ACR is always present.